zcbor/zcbor.py

```diff
@@ -2585,19 +2586,29 @@ def render_cmake_file(self, target_name, h_file, c_file, type_file, output_c_dir
     )
 """
 
-    def render(self, h_file, c_file, type_file, cmake_file=None, output_c_dir=None,
+    def render(self, h_file, c_file, type_file, include_dirs, cmake_file=None, output_c_dir=None,
                output_h_dir=None):
-        _, h_name = path.split(h_file.name)
 
         # Create and populate the generated c and h file.
         makedirs("./" + path.dirname(c_file.name), exist_ok=True)
 
+        if (include_dirs):
+            h_name = sorted(
+                [path.relpath(h_file.name, start=p.absolute()) for p in include_dirs],
+                key=lambda p: len(Path(p).parts))[0]
+            type_def_name = sorted(
+                [path.relpath(type_file.name, start=p.absolute()) for p in include_dirs],
+                key=lambda p: len(Path(p).parts))[0]
```

Comment:
These are a bit cryptic, and you do the same thing twice, if you pull out a private function and give it a good name I might understand what is happening.